src/plugins/console/public/application/models/sense_editor/__tests__/integration.test.js

```diff
@@ -661,7 +661,7 @@ describe('Integration', () => {
       {
         name: 'Any of - mixed - both',
         cursor: { lineNumber: 14, column: 3 },
-        autoCompleteSet: [tt('{'), tt(3)],
+        autoCompleteSet: [tt(3), tt('{')],
```

Comment:
The internal algorithm used by `Array.prototype.sort` has changed in Node.js v11, and the sorting algorithm we use to sort terms is unfortunately not immune to the order in which `Array.prototype.sort` checks each element.

In this case, the two terms are provided to the `sort` callback here as `t1` and `t2`. In Node.js 10, `t1` is `3` and `t2` is `'{'`, In Node.js 12 it's the opposite. In both cases, we end at the last `return` statement as both their scores are `0` and `3 < '{' === '{' < 3`.

https://github.com/elastic/kibana/blob/e16885c3ad2b3bf6aac3ba0a9dc0548dc5020741/src/plugins/console/public/lib/autocomplete/autocomplete.ts#L984-L1000

Reply:
Thanks for flagging and digging into this @watson , this test change seems innocuous enough for now. I'd want to hold of making your proposed change right now as this needs to be investigated a bit further so that we make sure the order of completions is consistent and in an order we are happy with.

I created this issue for tracking https://github.com/elastic/kibana/issues/83159